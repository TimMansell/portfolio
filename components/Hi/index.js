import Hi from './Hi';

export default Hi;
