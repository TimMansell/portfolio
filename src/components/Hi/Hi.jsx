import styles from './Hi.module.scss';

export const Hi = () => {
  return <div className={styles.hi}>Hi!</div>;
};

export default Hi;
