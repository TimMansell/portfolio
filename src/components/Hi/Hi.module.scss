@import 'variables';
@import 'utilities/pxtorem';

.hi {
  composes: hero shadowLg from '~scss/helpers/text.scss';
  font-size: pxtorem(128);
  letter-spacing: pxtorem(5);
  height: 0;
  margin: 0;
  animation: hi 3s ease-in-out;
  animation-fill-mode: forwards;

  @media (min-width: $bp-md) {
    font-size: pxtorem(160);
  }
}

@keyframes hi {
  35%,
  75% {
    opacity: 1;
    transform: translateY(0);
  }

  to {
    opacity: 0;
    transform: translateY(20px);
  }
}
