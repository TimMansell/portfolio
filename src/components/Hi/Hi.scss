@import 'variables';
@import 'utilities/breakpoints';

.hi {
	color: $secondary-text-colour;
	font-size: 80px;
	font-weight: 900;
	letter-spacing: 5px;
	line-height: 1;
	height: 0;
	margin: 0;
	text-transform: uppercase;
	text-shadow: 3px 3px darken($primary-colour, 10%);
	opacity: 0;

	animation: hi 3s ease-in-out;
	animation-delay: 1s;
	animation-fill-mode: forwards;

	@include media-breakpoint-up(md){
		text-shadow: 2px 2px $tertiary-text-colour;
		font-size: 100px;
	}
}

@keyframes hi {
	from {
		opacity: 0;
		transform: translateY(-10px);
	}

	35%, 65% {
		opacity: 1;
		transform: translateY(0);
	}

	to {
		transform: translateY(10px);
	}
}