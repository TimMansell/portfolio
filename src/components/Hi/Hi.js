import React from 'react';

import './Hi.scss';

export const Hi = () => {
	return <p className="hi">Hi!</p>;
}

export default Hi;