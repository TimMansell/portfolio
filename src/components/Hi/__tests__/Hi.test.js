import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Hi from '../Hi';

describe('Hi', () => {
  it('should render my component', () => {
    render(<Hi />);
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<Hi />);

    expect(asFragment()).toMatchSnapshot();
  });
});
