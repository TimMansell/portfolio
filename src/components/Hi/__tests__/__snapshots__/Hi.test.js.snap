// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hi should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="hi"
  >
    Hi!
  </div>
</DocumentFragment>
`;
